commons/all/src/main/java/org/infinispan/commons/configuration/io/yaml/YamlConfigurationReader.java

```diff
@@ -0,0 +1,366 @@
+                           namespaces.put("", parts[2]); // The primary namespace
+                        } else {
+                           namespaces.put(parts[1].substring(1, parts[1].length() - 1), parts[2]);
+                        }
+                        return parsed;
+                     } else if (parts.length == 2 && parts[0].equals("%YAML")) {
+                        return parsed;
+                     } else {
+                        Log.CONFIG.warn("Unknown directive " + s + " at " + Location.of(line, i));
+                     }
+                  }
+                  break;
+               case '-':
+                  if (i == 0) {
+                     if ("---".equals(s)) {
+                        // It's a separator
+                        return parsed;
+                     }
+                  } else if (s.charAt(i + 1) == ' ') {
+                     // It's an array delimiter
+                     parsed.list = true;
+                     i++;
+                     if (state == 0) {
+                        if (s.charAt(length - 1) != ':') {
+                           // It's a bare value
+                           state = 2;
+                        }
+                     }
+                  }
+                  break;
+               case '"':
+               case '\'':
+                  if (state == 0 || state == 2) {
+                     int endQuote = s.indexOf(c, i + 1);
+                     if (endQuote < 0) {
+                        throw new ConfigurationReaderException("Missing closing quote", Location.of(line, i));
+                     }
+                     String v = s.substring(i + 1, endQuote).trim();
+                     if (state == 0) {
+                        parseKey(parsed, v);
+                        state = 2;
+                     } else {
+                        parsed.value = v;
+                        state = 4;
+                     }
+                     i = endQuote + 1; // Skip
+                     break;
+                  }
+                  // Fallthrough
+               default:
+                  if (state == 0) {
+                     state = 1;
+                     start = i;
+                  } else if (state == 2) {
+                     state = 3;
+                     start = i;
+                  }
+            }
+         }
+         if (state == 1) { // Unterminated key
+            throw new ConfigurationReaderException("Incomplete line", Location.of(line, 1));
+         }
+         if (state == 3) { // we reached the end of the line
+            parsed.value = s.substring(start).trim();
+         }
+         return parsed;
+      }
+   }
+
+   private void parseKey(Parsed p, String s) {
+      int colon = s.lastIndexOf(':');
+      p.name = colon < 0 ? s : s.substring(colon + 1);
+      p.nsPrefix = colon < 0 ? "" : s.substring(0, colon);
+      if (!p.nsPrefix.isEmpty()) {
+         namespaces.putIfAbsent(p.nsPrefix, p.nsPrefix);
+      }
```

Comment:
WDYT to automatically converting camelCase to kebab-case on yaml keys?

This allows the yaml configuration to better resemble the operator k8s camelCase style. e.g.:

```yaml
authorization:
  auditLogger: org.infinispan.security.audit.NullAuditLogger
  identityRoleMapper: ~
  roles:
    - role:
      name: peasent
      permissions: READ
```

Reply:
YAML files will now use camelCase.